components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -203,4 +203,14 @@ private void setClientCredentialsFromParam(Map<String, List> params, OAuthClient
         context.setClientId(stringContent.get(OAuth.OAUTH_CLIENT_ID));
     }
 
+    /**
+     * Retrieve the referred name of the authenticator.
+     *
+     * @return      Referred name of the authenticator.
+     */
+    @Override
+    public String getReferredName() {
+        return "public_auth";
```

Comment:
Can we make these return as constants for future reference?

Reply:
Fixed in 684ce335e1241bfce1d6fd56aae4ca08abba580c